plugins/postcss-cascade-layers/src/desugar-and-parse-layer-names.ts

```diff
@@ -0,0 +1,121 @@
+import type { Container } from 'postcss';
+import type { Model } from './model';
+import selectorParser from 'postcss-selector-parser';
+import { INVALID_LAYER_NAME } from './constants';
+
+export function desugarAndParseLayerNames(root: Container, model: Model) {
+	// - parse layer names
+	// - rename anon layers
+	// - handle empty layers
+	root.walkAtRules('layer', (layerRule) => {
+		if (layerRule.params) {
+			const layerNameList: Array<string> = [];
+			let isInvalidLayerName = false;
+
+			selectorParser().astSync(layerRule.params).each((selector) => {
```

Comment:
```suggestion
			// Layer names ("A.B, C") have a similar syntax as selector lists with only class and tag name selectors.
			// - comma separated
			// - "." as a delimiter between idents.
			// We can use "postcss-selector-parser" to analyse layer names so that we don't have to implement our own tokenizer and parser.
			selectorParser().astSync(layerRule.params).each((selector) => {
```